lib/elixir/lib/macro.ex

```diff
@@ -2169,7 +2169,14 @@ defmodule Macro do
           case :elixir_config.identifier_tokenizer().tokenize(charlist) do
             {kind, _acc, [], _, _, special} ->
               if kind == :identifier and not :lists.member(?@, special) do
-                :identifier
+                # identifier_tokenizer used to return errors for non-nfc, but
+                #  now it nfc-normalizes everything. however, lack of nfc is
+                #  still a good reason to quote an atom when printing.
+                if charlist == :unicode.characters_to_nfc_list(charlist) do
```

Comment:
Good catch. However, instead of normalizing it again, we should return the relevant information from `tokenize`. I will see if there is a good way to include this information.

Reply:
Please ignore this comment for now. We need to tackle it but I am not sure how yet. :)

Reply:
Cool - in the interim, I appreciate the existing behavior now; quoting it is enough to indicate that I need to look at it closer ('this atom is quoted, so something is up with it'). But the UX for how/if to indicate something more specific, 'this atom is nfc-abnormal', when inspect()ing isn't something I have a good handle on. (And then there's also how to wind that info out, as you point out, from the tokenizer).

Aside -- I briefly, and wrongly, thought "oh, we should print out the normalized form now, since we we normalize atoms!" (That's wrong as this is an Atom that could have been generated from a String, as in the example in the tests, and we aren't normalizing all Atoms, just the ones that are literals in code).